internal/resources/aws/elastic_beanstalk_environment.go

```diff
@@ -0,0 +1,173 @@
+// as the default value for the "instances" usage param.  Without this, --sync-usage-file sets instances=0 causing the
+// costs for the node group to be $0.  This can be removed when --sync-usage-file creates the usage file with usgage keys
+// commented out by default.
+func (r *ElasticBeanstalkEnvironment) getUsageSchemaWithDefaultInstanceCount() []*schema.UsageItem {
+	var instanceCount = &r.instanceCount
+
+	if instanceCount == nil || *instanceCount == 0 {
+		return AutoscalingGroupUsageSchema
+	}
+
+	usageSchema := make([]*schema.UsageItem, 0, len(AutoscalingGroupUsageSchema))
+	for _, u := range AutoscalingGroupUsageSchema {
+		if u.Key == "instances" {
+			usageSchema = append(usageSchema, &schema.UsageItem{Key: "instances", DefaultValue: intVal(instanceCount), ValueType: schema.Int64})
+		} else {
+			usageSchema = append(usageSchema, u)
+		}
+	}
+	return usageSchema
+}
+
+// BuildResource builds a schema.Resource from a valid ElasticBeanstalkEnvironment struct.
+// This method is called after the resource is initialised by an IaC provider.
+// See providers folder for more information.
+func (r *ElasticBeanstalkEnvironment) BuildResource() *schema.Resource {
+
+	subResources := make([]*schema.Resource, 0)
+
+	rootBlockDevice := &EBSVolume{
+		Address: "aws_ebs_volume",
+		Region:  r.Region,
+		Type:    r.RootVolumeType,
+		Size:    r.RootVolumeSize,
+		IOPS:    r.RootVolumeIOPS,
+	}
+
+	launchConfiguration := &LaunchConfiguration{
+		Address:         "aws_launch_configuration",
+		Region:          r.Region,
+		InstanceType:    r.InstanceType,
+		InstanceCount:   &r.InstanceCount,
+		RootBlockDevice: rootBlockDevice,
+	}
+
+	autoScalingGroup := &AutoscalingGroup{
+		Address:             "aws_autoscaling_group",
+		Region:              r.Region,
+		Name:                r.Name,
+		LaunchConfiguration: launchConfiguration,
+	}
+
+	autoScalingGroupResource := autoScalingGroup.BuildResource()
+
+	var estimateInstanceQualities = autoScalingGroupResource.EstimateUsage
+
+	subResources = append(subResources, autoScalingGroupResource.SubResources...)
+	if r.LoadBalancerType == "classic" {
+		subResources = append(subResources, r.ElasticLoadBalancer.BuildResource())
+	} else {
+		subResources = append(subResources, r.LoadBalancer.BuildResource())
+	}
+
+	if r.StreamLogs {
+		subResources = append(subResources, r.CloudwatchLogGroup.BuildResource())
+	}
+
+	if r.RDSIncluded {
+		subResources = append(subResources, r.DBInstance.BuildResource())
+	}
+
+	estimate := func(ctx context.Context, u map[string]interface{}) error {
+		if estimateInstanceQualities != nil {
+			err := estimateInstanceQualities(ctx, u)
+			if err != nil {
+				return err
+			}
+		}
+		if r.Name != "" {
+			count, err := aws.AutoscalingGetInstanceCount(ctx, r.Region, r.Name)
+			if err != nil {
+				return err
+			}
+			if count > 0 {
+				u["instances"] = int64(math.Round(count))
+			}
+		}
+		return nil
+	}
```

Comment:
I don't believe we need this as we're not using any attributes for the `AutoscalingGroup` usage params. We could potentially add these in the future but I'm leaning towards not allowing them in at this point.